### PR TITLE
Adding snippet file for creating components via GraphQL

### PR DESCRIPTION
**Pull request by rvenkatesh-atl**
Adding a snippet to create components via GraphQL

### CODE REVIEW

File: snippets/graphql/create-component/README.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# How to use this
+
+This query will retrieve a list of metric definitions for your site.
```
Comment (sbland2) [inline snippets/graphql/create-component/README.md:3]: "This query will create a component."?

File: snippets/graphql/create-component/README.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+Replace `cloudId` below in the variables section with the cloudId for your site and execute the query. You can use [the GraphQL explorer](https://developer.atlassian.com/cloud/compass/graphql/explorer/) to run this query and explore [the Compass API](https://developer.atlassian.com/cloud/compass/graphql/) further.
```
Comment (sbland2) [inline snippets/graphql/create-component/README.md:5]: Maybe mention the "type" here as well and link to https://developer.atlassian.com/cloud/compass/components/what-is-a-component/#component-types

File: snippets/graphql/create-component/README.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+```
+{
+  "X-ExperimentalApi": ["compass-beta","compass-prototype"]
+}
+```
+
+### Query Variables
+
+Type field could include following components - Service, Library, Application or Other. We are using "service" in the example below
```
Comment (eserranovalle) [inline snippets/graphql/create-component/README.md:34]: Is the type case-sensitive? Even if the example is giving an idea that it actually is, we could include it them in the same case in this text (e.g.: `SERVICE`).

File: snippets/graphql/create-component/README.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+```graphql
+mutation createComponent($cloudId: ID!, $componentDetails: CreateCompassComponentInput!) {
+  compass {
+    createComponent(cloudId: $cloudId, input: $componentDetails) {
+      success
+      componentDetails {
+        id
+        name
+        type
```
Comment (eserranovalle) [inline snippets/graphql/create-component/README.md:17]: When I run this query, it marks `type` as deprecated in favor of `typeId`. The query still works, though.

**Review by eserranovalle — APPROVED** [review]

File: snippets/graphql/create-component/README.md (hunk 3 above)
Comment (rvenkatesh-atl) [reply snippets/graphql/create-component/README.md:34]: solid pointer, yes!

File: snippets/graphql/create-component/README.md (hunk 4 above)
Comment (rvenkatesh-atl) [reply snippets/graphql/create-component/README.md:17]: good point, i should translate this back to typeID, we introduced that recently.

File: snippets/graphql/create-component/README.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+### Query Variables
+
+We are using "SERVICE" as a component type in the example below, to learn about different component types, check this [link](https://developer.atlassian.com/cloud/compass/components/what-is-a-component/#component-types)
+
+```
+{
+      "cloudId": "your-cloud-id",
+      "componentDetails": {
+          "name": "<XYZ>",
+          "type": "SERVICE"
```
Comment (sbland2) [inline snippets/graphql/create-component/README.md:41]: `typeId` here as well

**Review by sbland2 — APPROVED** [review]

**Review by eserranovalle — APPROVED** [review]

**Review by sbland2 — APPROVED** [review]

**Review by eserranovalle — APPROVED** [review]

**Pull request merged**